### PR TITLE
feat: Add ability to use env names of TF Azure provider

### LINKED ISSUE #106: Ability to use terraform style variables for SPN auth
Hi there 👋

We at @swisspost really like kubelogin. Would it be possible to implement a method to use it during a terraform run and reusing the env variables set for the terraform-provider-azurerm:

> ```console
> $ export ARM_CLIENT_ID="00000000-0000-0000-0000-000000000000"
> $ export ARM_CLIENT_SECRET="00000000-0000-0000-0000-000000000000"
> $ export ARM_SUBSCRIPTION_ID="00000000-0000-0000-0000-000000000000"
> $ export ARM_TENANT_ID="00000000-0000-0000-0000-000000000000"
> ```
> _Source: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/service_principal_client_secret#configuring-the-service-principal-in-terraform_

Are you open for such a feature (PR)?

/xref:
- https://github.com/hashicorp/terraform-provider-kubernetes/issues/1753

### LINKED ISSUE #109: Missing env for "AAD tenant ID" in `spn` login method
Hi

When using `spn` login method, the following arguments are required:
- `--client-id` (also available as an env)
- `--client-secret` (also available as an env)
- `--server-id`
- `--tenant-id` (only available as env when using `msi` login method)

I think we should move the env outside the conditional and rename the variable to a generic name

```diff
+       if v, ok := os.LookupEnv(envTenantID); ok {
+               o.TenantID = v
+       }
+
        if o.LoginMethod == WorkloadIdentityLogin {
                if v, ok := os.LookupEnv(envWorkloadIdentityClientID); ok {
                        o.ClientID = v
                }
-               if v, ok := os.LookupEnv(envWorkloadIdentityTenantID); ok {
-                       o.TenantID = v
-               }
```

Is this a valable feature request? I'd file a PR for this.

### PR DESCRIPTION
**Pull request by mkilchhofer**
Resolves #106
Resolves #109 

Links to issues:
- #106 
- #109

### CODE REVIEW

File: pkg/token/options.go [hunk 1]
```diff
@@ -10,21 +10,22 @@ import (
...
+	ServerID               string
+	TenantID               string
+	Environment            string
+	IsLegacy               bool
+	TokenCacheDir          string
+	tokenCacheFile         string
+	IdentityResourceId     string
+	FederatedTokenFile     string
+	AuthorityHost          string
+	UseAzurermTerraformEnv bool
```
Comment (weinong) [inline pkg/token/options.go:28]:
```suggestion
	UseAzureRMTerraformEnv bool
```

**Comment by weinong** [comment]
LGTM. please rebase, though.

**Comment by mkilchhofer** [comment]
> LGTM. please rebase, though.

@weinong thanks for the review. Rebase done 👍

**Review by weinong — APPROVED** [review]
LGTM

**Pull request merged**